# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/FinalizationRegistryObject.cpp
# Commit: ac6d26908599
# Full Hash: ac6d26908599e15b1c129cbb0f252ca9b82836c6
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-07-03 03:56:55
# Description:
#   Bug 1648901 - Fix crash tracing partially constructed FinalizationRegistrationsObject r=sfink
#   
#   The shell allocation metadata hook means we can see these before they've been fully initialised.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D81521
# ==============================================================================

diff -r 21569aa4bced -r ac6d26908599 js/src/builtin/FinalizationRegistryObject.cpp
--- a/js/src/builtin/FinalizationRegistryObject.cpp	Mon Jun 29 18:45:45 2020 +0000
+++ b/js/src/builtin/FinalizationRegistryObject.cpp	Tue Jun 30 22:31:29 2020 +0000
@@ -184,8 +184,10 @@
   }
 
   auto* self = &obj->as<FinalizationRegistrationsObject>();
-  TraceRange(trc, self->records()->length(), self->records()->begin(),
-             "FinalizationRegistrationsObject records");
+  if (WeakFinalizationRecordVector* records = self->records()) {
+    TraceRange(trc, records->length(), records->begin(),
+               "FinalizationRegistrationsObject records");
+  }
 }
 
 /* static */